Quarkus Gradle plugin fails to initialize when an "aes-gcm-nopadding"-encrypted value is present in application.properties

### Describe the bug

The Gradle Quarkus plugin gives the following error if any `${aes-gcm-nopadding::...}` values are present in the application.properties file:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':quarkusBuild'.
> SRCFG00046: Could not find a secret key handler for aes-gcm-nopadding
```

The build.gradle file has the correct dependencies:

```gradle
dependencies {
    implementation enforcedPlatform("${quarkusPlatformGroupId}:${quarkusPlatformArtifactId}:${quarkusPlatformVersion}")
    implementation 'io.quarkus:quarkus-arc'
    implementation 'io.smallrye.config:smallrye-config-crypto'
    testImplementation 'io.quarkus:quarkus-junit5'
}
```

This failure does not appear in a Maven build.

Here is a portion of the stack trace:
```
Caused by: java.util.NoSuchElementException: SRCFG00046: Could not find a secret key handler for aes-gcm-nopadding
        at io.smallrye.config.SecretKeysHandlerConfigSourceInterceptor.getSecretValue(SecretKeysHandlerConfigSourceInterceptor.java:35)
        at io.smallrye.config.SecretKeysHandlerConfigSourceInterceptor.getValue(SecretKeysHandlerConfigSourceInterceptor.java:24)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
        at io.smallrye.config.FallbackConfigSourceInterceptor.getValue(FallbackConfigSourceInterceptor.java:24)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
        at io.smallrye.config.PropertyNamesConfigSourceInterceptor.getValue(PropertyNamesConfigSourceInterceptor.java:17)
        at io.smallrye.config.SmallRyeConfigSourceInterceptorContext.proceed(SmallRyeConfigSourceInterceptorContext.java:20)
        at io.smallrye.config.SmallRyeConfig.getConfigValue(SmallRyeConfig.java:323)
        at io.quarkus.gradle.tasks.EffectiveConfig.lambda$generateFullConfigMap$1(EffectiveConfig.java:108)
        at io.quarkus.gradle.tasks.EffectiveConfig.generateFullConfigMap(EffectiveConfig.java:107)
        at io.quarkus.gradle.tasks.EffectiveConfig.<init>(EffectiveConfig.java:87)
        at io.quarkus.gradle.tasks.EffectiveConfig$Builder.build(EffectiveConfig.java:165)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.buildEffectiveConfiguration(AbstractQuarkusExtension.java:128)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.buildBaseConfig(AbstractQuarkusExtension.java:69)
        at org.gradle.api.internal.provider.DefaultProvider.calculateOwnValue(DefaultProvider.java:72)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:108)
        at org.gradle.api.internal.provider.TransformBackedProvider.calculateOwnValue(TransformBackedProvider.java:82)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.calculateValue(AbstractMinimalProvider.java:108)
        at org.gradle.api.internal.provider.AbstractMinimalProvider.withFinalValue(AbstractMinimalProvider.java:164)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:133)
        at org.gradle.api.internal.provider.DefaultProperty.finalValue(DefaultProperty.java:26)
        at org.gradle.api.internal.provider.AbstractProperty.finalizeNow(AbstractProperty.java:245)
        at org.gradle.api.internal.provider.AbstractProperty.finalizeValue(AbstractProperty.java:194)
        at io.quarkus.gradle.tasks.AbstractQuarkusExtension.baseConfig(AbstractQuarkusExtension.java:74)
        at io.quarkus.gradle.tasks.QuarkusBuildTask.packageType(QuarkusBuildTask.java:60)
        at io.quarkus.gradle.tasks.QuarkusBuild.getBuildInputFiles(QuarkusBuild.java:168)
        at io.quarkus.gradle.tasks.QuarkusBuild_Decorated.getBuildInputFiles(Unknown Source)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.annotations.impl.DefaultPropertyAnnotationMetadata.getPropertyValue(DefaultPropertyAnnotationMetadata.java:97)
        at org.gradle.internal.properties.annotations.DefaultTypeMetadataStore$DefaultPropertyMetadata.getPropertyValue(DefaultTypeMetadataStore.java:266)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$1.lambda$visitLeaf$0(DefaultPropertyWalker.java:88)
        at org.gradle.internal.deprecation.DeprecationLogger.whileDisabled(DeprecationLogger.java:258)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$CachedPropertyValue.lambda$new$0(DefaultPropertyWalker.java:105)
        at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183)
        at org.gradle.internal.properties.bean.DefaultPropertyWalker$CachedPropertyValue.call(DefaultPropertyWalker.java:147)
        at org.gradle.util.internal.GUtil.uncheckedCall(GUtil.java:437)
        at org.gradle.util.internal.DeferredUtil.unpackNestableDeferred(DeferredUtil.java:83)
        at org.gradle.api.internal.file.collections.UnpackingVisitor.add(UnpackingVisitor.java:90)
        at org.gradle.api.internal.file.DefaultFileCollectionFactory$ResolvingFileCollection.visitChildren(DefaultFileCollectionFactory.java:285)
        at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:122)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:103)
        at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmptyEdges.getNodeValues(CachingDirectedGraphWalker.java:213)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:121)
        at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:73)
        at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:66)
        ... 142 more
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Reproducer: https://github.com/jskillin-idt/quarkusio-quarkus-issues-33135

To recreate the reproducer:
1. Simply create a repo: `quarkus create app --no-code --gradle`
2. In `src/main/resources/application.properties` add the line: `someValue = ${aes-gcm-nopadding::garbage}` (it doesn't matter what it is as long as it references "aes-gcm-nopadding")

With either your own repo or the reproducer, simply:
1. Run `./gradlew build`

Notice that the configuration of the build.gradle, through dependencies or classpath, does not appear to affect the outcome.

### Output of `uname -a` or `ver`

Linux jskillin-ubuntu-dev 6.2.0-20-generic #20-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  6 07:48:48 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.18" 2023-01-17 OpenJDK Runtime Environment (build 11.0.18+10-post-Ubuntu-0ubuntu5) OpenJDK 64-Bit Server VM (build 11.0.18+10-post-Ubuntu-0ubuntu5, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.0.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.1

### Additional information

_No response_

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -56,7 +56,7 @@
         <microprofile-lra.version>2.0</microprofile-lra.version>
         <microprofile-openapi.version>3.1</microprofile-openapi.version>
         <smallrye-common.version>2.1.0</smallrye-common.version>
-        <smallrye-config.version>3.2.1</smallrye-config.version>
+        <smallrye-config.version>3.3.0</smallrye-config.version>
         <smallrye-health.version>4.0.2</smallrye-health.version>
         <smallrye-metrics.version>4.0.0</smallrye-metrics.version>
         <smallrye-open-api.version>3.3.4</smallrye-open-api.version>

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/EffectiveConfig.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/tasks/EffectiveConfig.java
@@ -116,6 +116,7 @@ final class EffectiveConfig {
     @VisibleForTesting
     static SmallRyeConfig buildConfig(String profile, List<ConfigSource> configSources) {
         return ConfigUtils.emptyConfigBuilder()
+                .setAddDiscoveredSecretKeysHandlers(false)
                 .withSources(configSources)
                 .withProfile(profile)
                 .build();

--- a/integration-tests/smallrye-config/src/test/java/io/quarkus/it/smallrye/config/ServerResourceTest.java
+++ b/integration-tests/smallrye-config/src/test/java/io/quarkus/it/smallrye/config/ServerResourceTest.java
@@ -102,6 +102,6 @@ class ServerResourceTest {
                 .body("errors", hasItem("cloud.info.admins.root[1].username size must be between 0 and 4"))
                 .body("errors", hasItem("cloud.info.firewall.accepted[1] size must be between 8 and 15"))
                 .body("errors", hasItem("cloud.proxy.timeout must be less than or equal to 10"))
-                .body("errors", hasItem("server is not prod"));
+                .body("errors", hasItem("cloud server is not prod"));
     }
 }

